File: Java/app/src/main/java/com/tyhoo/mvp/java/ui/MainContract.java
package com.tyhoo.mvp.java.ui;

import com.tyhoo.mvp.java.base.BasePresenter;
import com.tyhoo.mvp.java.base.BaseView;
import com.tyhoo.mvp.java.data.User;

public interface MainContract {

    interface View extends BaseView<Presenter> {

        void setLoadingIndicator(boolean active);

        void showUser(String user);

        void hideUser();

        void showParseJson(User user);
    }

    interface Presenter extends BasePresenter {

        void parseJson();
    }
}

File: Kotlin/app/src/main/java/com/tyhoo/mvp/kotlin/ui/MainContract.kt
package com.tyhoo.mvp.kotlin.ui

import com.tyhoo.mvp.kotlin.base.BasePresenter
import com.tyhoo.mvp.kotlin.base.BaseView
import com.tyhoo.mvp.kotlin.data.User

/**
 * This specifies the contract between the view and the presenter.
 */
interface MainContract {

    interface View : BaseView<Presenter> {

        fun setLoadingIndicator(active: Boolean)

        fun showUser(user: String)

        fun hideUser()

        fun showParseJson(user: User)
    }

    interface Presenter : BasePresenter {

        fun parseJson()
    }
}